st2reactor/st2reactor/garbage_collector/executions.py

```diff
@@ -0,0 +1,105 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Module with utility functions for purging old action executions and
+corresponding live action objects.
+"""
+
+import copy
+
+from mongoengine.errors import InvalidQueryError
+
+from st2common.constants import action as action_constants
+from st2common.persistence.liveaction import LiveAction
+from st2common.persistence.execution import ActionExecution
+
+__all__ = [
+    'purge_executions'
+]
+
+DONE_STATES = [action_constants.LIVEACTION_STATUS_SUCCEEDED,
+               action_constants.LIVEACTION_STATUS_FAILED,
+               action_constants.LIVEACTION_STATUS_TIMED_OUT,
+               action_constants.LIVEACTION_STATUS_CANCELED]
+
+
+def purge_executions(logger, timestamp, action_ref=None, purge_incomplete=False):
```

Comment:
Why is passing in the logger a good idea. This takes away from the interface of this method and  adds very little value.